### ISSUE TITLE
[Bug]: Peeping Dorm Manager MonoBehaviours

### ISSUE THREAD

**Issue by LuckyMenOrNot** [AssetRipper/AssetRipper#1212]
### Are you on the latest version of AssetRipper?

Yes, I'm on the latest release of AssetRipper.

### Which release are you using?

Windows x64

### Which game did this occur on?

https://store.steampowered.com/app/2273420/Peeping_Dorm_Manager/

### Which Unity version did this occur on?

![image](https://github.com/AssetRipper/AssetRipper/assets/97903078/6e6c858f-311f-4a49-b4b4-589ef6523970)

### Is the game Mono or IL2Cpp?

IL2Cpp

### Describe the issue.

![image](https://github.com/AssetRipper/AssetRipper/assets/97903078/fa1eb71e-7c15-40a9-b2d8-9c1cb1a85d34)

### Relevant log output

[AssetRipper.log](https://github.com/AssetRipper/AssetRipper/files/14389167/AssetRipper.log)

**Comment by JMVRy** [comment]
TL;DR: the error logs just mean that it couldn't get the MonoBehaviours, which is pretty common for most tools. As long as the game assets were exported intact, then it should be fine. Use a different tool if you want to get the scripts.

<br />

That seems to happen no matter what (IL2Cpp or not, different tools as well), but it doesn't seem to affect any other assets, so as long as you get any other assets without issue, that shouldn't be a problem. You won't be able to get scripts back using this tool, which is informed on the Settings page: "Default. This exports full methods for Mono games and dummy methods for IL2Cpp games."

If you want the scripts decompiled, you may need to use another tool like `Cpp2IL`, `IL2CppInspector`, or even a tool like Ghidra, IDA, or x64dbg. The first 2 tools will turn the IL2Cpp back into C# DLLs, which can be inspected by another tool like `dnSpy` or `ILSpy`. The other two will decompile the IL2Cpp code back into readable C++ text.

**Comment by ds5678** [comment]
> it couldn't get the MonoBehaviours

That is incorrect. It could not read the fields for these particular MonoBehaviours. There are often tens of thousands in a game, so this is a very small failure rate.

**Comment by ds5678** [comment]
Other than #698, the MonoBehaviour issues in this game have been resolved, presumably by my recent commits.
